packages/ui/components/input-file/src/LionInputFile.js

```diff
@@ -0,0 +1,879 @@
+      this._syncAriaDescribedByAttributes();
+    }
+
+    /**
+     * Update _uploadedFilesMetaData only if:
+     *   1. It is invoked from the file-removed event handler.
+     *   2. There is a mismatch between the uploaded files and files on UI.
+     */
+    if (changedProperties.has('uploadResponse')) {
+      if (this.uploadResponse.length && this._uploadedFilesMetaData.length === 0) {
+        this.uploadResponse.forEach(preResponse => {
+          const file = {
+            systemFile: {
+              name: preResponse.name,
+            },
+            response: preResponse,
+            status: preResponse.status,
+            validationFeedback: [
+              {
+                message: preResponse.errorMessage,
+              },
+            ],
+          };
+          // @ts-ignore
+          this._uploadedFilesMetaData = [...this._uploadedFilesMetaData, file];
+        });
+      }
+      this._uploadedFilesMetaData.forEach(file => {
+        if (
+          !this.uploadResponse.some(response => response.name === file.systemFile.name) &&
+          !this.uploadOnFormSubmit
+        ) {
+          this.removeFileFromList(file);
+        } else {
+          this.uploadResponse.forEach(response => {
+            if (response.name === file.systemFile.name) {
+              // eslint-disable-next-line no-param-reassign
+              file.response = response;
+              // eslint-disable-next-line no-param-reassign
+              file.downloadUrl = response.downloadUrl ? response.downloadUrl : file.downloadUrl;
+              // eslint-disable-next-line no-param-reassign
+              file.status = response.status;
+              // eslint-disable-next-line no-param-reassign
+              file.validationFeedback = [
+                {
+                  type: response.errorMessage?.length > 0 ? 'error' : 'success',
+                  message: response.errorMessage,
+                },
+              ];
+            }
+          });
+          this._uploadedFilesMetaData = [...this._uploadedFilesMetaData];
+        }
+      });
+      this.requestUpdate();
+    }
+  }
+
+  /**
+   * @param {InputFile[]} fileList
+   */
+  computeNewAddedFiles(fileList) {
+    const computedFileList = fileList.filter(
+      file =>
+        this._uploadedFilesMetaData.findIndex(
+          existLionFile => existLionFile.systemFile.name === file.name,
+        ) === -1,
+    );
+    const duplicateFileNamesValidator = this.validators.find(
+      vali => vali instanceof DuplicateFileNames,
+    );
+    if (duplicateFileNamesValidator) {
+      if (fileList.length !== computedFileList.length) {
+        duplicateFileNamesValidator.param = { show: true };
+        this.validate();
+      } else {
+        duplicateFileNamesValidator.param = { show: false };
+        this.validate();
+      }
+    }
+    return computedFileList;
+  }
+
+  /**
+   * @param {DragEvent} ev
+   */
+  uploadDroppedFiles(ev) {
```

Comment:
This method does not actually upload the files to a server and is therefore a bit misleading. Should we rename it to something like `processDroppedFiles` or `parseDroppedFiles`?

Reply:
renamed